hazelcast/src/main/java/com/hazelcast/internal/util/Clock.java

```diff
@@ -112,7 +114,7 @@ protected long toSystemCurrentTimeMillis(long millis) {
 
         @Override
         protected long currentTimeMillis() {
-            return System.currentTimeMillis();
+            return TimeUnit.NANOSECONDS.toMillis(System.nanoTime());
```

Comment:
This is not correct. The nanotime starts with some arbitrary value and not the epoch time like System.currentTimeMillis.

Nanotime can only be used with time delta's and need to be treated specially to prevent overflow problems.